### PR TITLE
Fix window related bugs

### PR DESCRIPTION
**Pull request by Flamefire**
- Improve error messages for SDL on windows (https://www.siedler25.org/index.php?com=forum&mod=forum&action=thread&id=1583&start=-1#last)
- Fix auto version update #1008
- Refactor some window and window manager stuff
- Fix SDL2 scroll direction #1039 
- Fix localization of text files and improve its performance (lag on open) #1040

### CODE REVIEW

File: extras/videoDrivers/SDL/VideoSDL.cpp [hunk 1]
```diff
@@ -45,7 +47,29 @@ struct DeleterDeleteRC
...
+{
+    std::unique_ptr<HGLRC, DeleterDeleteRC> context{wglCreateContext(deviceContext)};
+    if(!context)
+        throw makeLastSystemError("Failed to create OGL context");
+    return context;
+}
+void ShareOGLResources(HGLRC srcContext, HGLRC dstContext)
+{
+    if(!wglShareLists(srcContext, dstContext))
+        throw makeLastSystemError("Failed to create OGL context");
```
Comment (Flow86) [inline extras/videoDrivers/SDL/VideoSDL.cpp]: copy and paste. I think here there should be a "Failed to share OGL lists".

Comment (Flamefire) [reply extras/videoDrivers/SDL/VideoSDL.cpp]: Yes thx. We do have a problem though: We need Boost build with fPIC if we want to use it in the drivers. For the purpose of `makeLastSystemError` I could implement that myself, but if we need more of boost we need the fPIC or shared boost libs.

Comment (Flamefire) [reply extras/videoDrivers/SDL/VideoSDL.cpp]: Fixing this by using std::error_code

**Comment by Flamefire** [comment]
@Flow86 Ready to merge

File: extras/videoDrivers/SDL/VideoSDL.cpp [hunk 2]
```diff
@@ -20,10 +20,12 @@
 #include "driver/VideoDriverLoaderInterface.h"
 #include "driver/VideoInterface.h"
 #include "helpers/containerUtils.h"
-#include <memory>
+#include "makeException.h"
 #include <boost/nowide/iostream.hpp>
+#include <boost/system/windows_error.hpp>
```
Comment (Flow86) [inline extras/videoDrivers/SDL/VideoSDL.cpp]: why is the windows_error.hpp included here? I can't see anything related with it?

File: libs/common/src/makeException.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include "makeException.h"
+#include <boost/system/config.hpp>
+#ifdef BOOST_WINDOWS_API
+#include <windows.h>
+#elif defined(BOOST_POSIX_API)
+#include <errno.h>
```
Comment (Flow86) [inline libs/common/src/makeException.cpp]: shouldn't it be 'cerrno'? and why not removing boost here completly and use `_WIN32` instead?

File: libs/rttrConfig/CMakeLists.txt [hunk 4]
```diff
@@ -17,20 +17,9 @@ SET_PROPERTY(TARGET rttrConfig PROPERTY POSITION_INDEPENDENT_CODE ON)
...
-        message(FATAL_ERROR "Could not get git revision. The source has to be in a git repo or you have to define RTTR_REVISION")
-    endif()
-    set(RTTR_REVISION ${RTTR_REVISION} CACHE STRING "Git revision" FORCE)
-endif()
-if(NOT RTTR_VERSION)
-    STRING(TIMESTAMP RTTR_VERSION "%Y%m%d")
-    set(RTTR_VERSION ${RTTR_VERSION} CACHE STRING "Version (yyyymmdd)" FORCE)
+# This should happen in the top-level CMakeLists
+if(NOT RTTR_REVISION OR NOT RTTR_VERSION)
+    message(FATAL_ERROR "Internal error: RTTR_REVISION or RTTR_VERSION not set")
```
Comment (Flow86) [inline libs/rttrConfig/CMakeLists.txt]: we should add a "in the top level CMakeList.txt" to the message

Comment (Flamefire) [reply libs/rttrConfig/CMakeLists.txt]: The comment above already says that. Should be enough

File: extras/videoDrivers/SDL/VideoSDL.cpp (hunk 2 above)
Comment (Flamefire) [reply extras/videoDrivers/SDL/VideoSDL.cpp]: Removed

File: libs/common/src/makeException.cpp (hunk 3 above)
Comment (Flamefire) [reply libs/common/src/makeException.cpp]: BEcause this should also be used for cygwin. The boost define has both although we could do it manually.

Comment (Flow86) [reply libs/common/src/makeException.cpp]: ah yes I don't know if cygwin defines _WIN32 and since we're using boost anyway ...

**Pull request merged**